anaAllChambersThresh.py

```diff
@@ -0,0 +1,142 @@
+
+    cmd1 = ["python","%s/vfatqc-python-scripts/macros/anaUltraThreshold.py"%(os.getenv("BUILD_HOME"))]
+    cmd1.append("--infilename=%s"%(filename_thresh))
+    cmd1.append("--chConfigKnown")
+    cmd1.append("--fileScurveFitTree=%s"%(filename_scurve))
+    cmd1.append("--vfatmask=0x0")
+
+    cmd2 = ["mkdir","-p","%s"%(elogPath)]
+    cmd3 = ["cp","%s/%s/threshold/channel/%s/ThresholdScanData/ThreshPrunedSummary.png"%(dataPath,cName,scandate_thresh),
+            "%s/ThreshPrunedSummary_%s.png"%(elogPath,cName)]
+    cmd4 = ["cp","%s/%s/threshold/channel/%s/ThresholdScanData/ThreshSummary.png"%(dataPath,cName,scandate_thresh),
+            "%s/ThreshSummary_%s.png"%(elogPath,cName)]
+    cmd5 = ["cp","%s/%s/threshold/channel/%s/ThresholdScanData/VFATPrunedSummary.png"%(dataPath,cName,scandate_thresh),
+            "%s/VFATPrunedSummary_%s.png"%(elogPath,cName)]
+    cmd6 = ["cp","%s/%s/threshold/channel/%s/ThresholdScanData/VFATSummary.png"%(dataPath,cName,scandate_thresh),
+            "%s/VFATSummary_%s.png"%(elogPath,cName)]    
+    cmd7 = ["cp","%s/%s/threshold/channel/%s/ThresholdScanData/VT1MaxSummary.png"%(dataPath,cName,scandate_thresh),
+            "%s/VT1MaxSummary_%s.png"%(elogPath,cName)]
+    try:
+        runCommand(cmd1)
+        runCommand(cmd2)
+        runCommand(cmd3)
+        runCommand(cmd4)
+        runCommand(cmd5)
+        runCommand(cmd6)
+        runCommand(cmd7)
+    except CalledProcessError as e:
+        print "Caught exception",e
+        pass
+    return
+
+
+if __name__ == '__main__':
+
+    import sys,os,signal
+    import subprocess
+    import itertools
+    from multiprocessing import Pool, freeze_support
+    from chamberInfo import chamber_config,GEBtype
+    from gempython.utils.wrappers import envCheck
+    import glob
+
+    from qcoptions import parser
+
+    parser.add_option("--scandate", type="string", dest="scandate", default="current",
+                      help="Specify specific date to analyze", metavar="scandate")
+    parser.add_option("--scandateTrim", type="string", dest="scandateTrim", default="current",
+                      help="Specify specific date of trim run used to generate chConfig used for --scandate's threshold scan", metavar="scandateTrim")
+
+    (options, args) = parser.parse_args()
+
+    envCheck('DATA_PATH')
+    envCheck('ELOG_PATH')
+    
+    #Look for threshold scan directory
+    searchPath  = "%s/GEMINI*/threshold/channel/*"%(os.getenv('DATA_PATH'))
+    dirs        = glob.glob(searchPath)
+    foundDir    = False
+
+    for path in dirs:
+        if path.rfind(options.scandate) > 0:
+            foundDir = True
+            pass
+        pass
+    if not foundDir:
+        print "Unable to find %s in output location specified: %s"%(options.scandate,searchPath)
+        exit(50)
+    else:
+        print "Found %s"%(options.scandate)
+        pass
+    
+    #Look for trim directory
+    searchPath  = "%s/GEMINI*/scurve/*"%(os.getenv('DATA_PATH'))
```

Comment:
I am looking for the directory in which the TFile containing "scurveFitTree" is located.  This should correspond to the chConfig.txt file that was used in "confChamber.py" to setup the trimDAC's before running the threshold scan.

Explicitly if I wanted to analyze the threshold scan data taken:

http://cmsonline.cern.ch/cms-elog/991007

I would run with the scurveFitTree files obtained: http://cmsonline.cern.ch/cms-elog/990812